topcoffea/modules/YieldTools.py

```diff
@@ -415,32 +612,48 @@ def get_normalized_yield(self,hin_dict,year,proc,cat_dict,overflow_str,rwgt_pt=N
     #   - Takes as input a hist dict (i.e. what the processor outptus)
     #   - Returns a dictionary of yields for the categories in the "channel" axis
     #   - Optionally sums over njets or lep flavs
-    def get_yld_dict(self,hin_dict,year,njets=False,lepflav=False):
+    def get_yld_dict(self,hin_dict,year=None,njets=False,lepflav=False):
 
         # Check for options that do not make sense
         if lepflav and not self.is_split_by_lepflav(hin_dict):
             raise Exception("Error: Cannot split by lep flav if the input file is not split by lep flav.")
 
         # If we want to seperate by njets, don't use njets hist since njets are not in it's sparse axis
         hist_to_use = "njets"
-        if njets: hist_to_use = "ht"
+        #if njets: hist_to_use = "ht"
+        if njets: hist_to_use = "lj0pt"
 
         # Get the cat dict (that we will integrate over)
         cat_dict = {}
         for ch in self.get_cat_lables(hin_dict,"channel",h_name=hist_to_use):
             cat_dict[ch] = {}
             nlep_str = ch.split("_")[0]
-            cat_dict[ch]["appl"] = self.APPL_DICT[nlep_str]
+            if "appl" in self.get_axis_list(hin_dict[hist_to_use]):
+                cat_dict[ch]["appl"] = self.APPL_DICT[nlep_str]
             cat_dict[ch]["channel"] = ch
 
         # Find the yields
         yld_dict = {}
         proc_lst = self.get_cat_lables(hin_dict,"sample")
+        if "flipsUL17" not in proc_lst: proc_lst = proc_lst + ["flipsUL16","flipsUL16APV","flipsUL17","flipsUL18"] # Very bad workaround for _many_ reasons
+        print("proc_lst",proc_lst)
+        #for proc in proc_lst:
+        #    p = self.get_short_name(proc)
+        #    print("Name:",p,proc) # Print what name the sample has been matched to
```

Comment:
Remove?

Reply:
Uncommented (I run with this uncommented more than commented, so probably better to uncomment it than to remove it)